tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddGradeItemPanelContent.java

```diff
@@ -22,24 +18,14 @@
 
 	private static final long serialVersionUID = 1L;
 
-	public AddGradeItemPanelContent(String id, Form form) {
-        this(id, form, "", "");
-	}
-
-    public AddGradeItemPanelContent(String id, Form form, String title, String points) {
+	public AddGradeItemPanelContent(String id) {
         super(id);
 
-        Assignment assignment = new Assignment();
-        assignment.setName(title);
-        if (points != null && !"".equals(points))
-            assignment.setPoints(Double.parseDouble(points));
-
-        add(new TextField("title", new PropertyModel<String>(assignment, "name")));
-        add(new TextField("points", new PropertyModel<Double>(assignment, "points")));
-        add(new DateTextField("duedate", new PropertyModel<Date>(assignment, "dueDate"), "MM/DD/yy"));
+        add(new TextField("name"));
```

Comment:
I think for code clarity we should explicity bind the Assignment properties onto these fields, even though its implicit.
So:
add(new TextField("name", assignmentModel.bind("name"));

you would need to get the model before you did that since it is created in the superclass.
This also means that the first 'name' doesnt have to match exactly the Assignment name field, so if someone changes the names of these fields in the future we don't need to change both the HTML and the code.

The other thing is I think we should use our own page based model for this, then in the onSubmit, copy the values over from our model to the gradebook.shared.Assignment object to create the assignment. This gives us some more flexibility if we want to add more fields or change things around a bit. We do have some screens coming up where things in the old GB are combined so using the GB objects wont work.